src/sdk/propagation/baggage.rs

```diff
@@ -76,10 +76,16 @@ impl TextMapPropagator for BaggagePropagator {
         if !baggage.is_empty() {
             let header_value = baggage
                 .iter()
-                .map(|(name, value)| {
+                .map(|(name, (value, metadata))| {
+                    let metadata_str = if metadata.as_str().trim().is_empty() {
+                        "".to_string()
```

Comment:
this can probably have `metadata_str` be a str instead of string, don't need either string conversion or the as str below

Reply:
Hmm I think in `else` caluse. The `format!(";{}"...)` cannot be stored as `&str` since it will be freed at the end clause if `metadata_str` doesn't own it.

Reply:
oh right you need the prefix. What about something like:

```rust
let metadata_str = metadata.as_str().trim();
let prefix = if metadata_str.is_empty() { "" } else { ";" };
utf8_percent_encode(name.as_str().trim(), FRAGMENT)
    .chain(iter::once("="))
    .chain(utf8_percent_encode(String::from(value).trim(), FRAGMENT))
    .chain(iter::once(prefix))
    .chain(iter::once(metadata_str))
    .collect()
})
```

Reply:
Great idea :+1: